core/src/main/java/io/kestra/core/models/triggers/types/Schedule.java

```diff
@@ -196,56 +200,64 @@ public ZonedDateTime nextEvaluationDate(ConditionContext conditionContext, Optio
             }
 
             // previous present but no scheduleConditions
-            return computeNextEvaluationDate(executionTime, lastDate).orElse(null);
+            nextDate = computeNextEvaluationDate(executionTime, lastDate).orElse(null);
         }
 
-
         // no previous present but backfill
-        if (backfill != null && backfill.getStart() != null) {
-            return this.timezone != null ?
+        else if (backfill != null && backfill.getStart() != null) {
+            nextDate = this.timezone != null ?
                 backfill.getStart().withZoneSameLocal(ZoneId.of(this.timezone)) :
                 backfill.getStart();
+
+            // if max delay reach, we calculate a new date
+            if (this.lateMaximumDelay != null) {
+                ScheduleDates scheduleDates = this.scheduleDates(executionTime, nextDate).orElse(null);
+                scheduleDates = this.handleMaxDelay(scheduleDates);
+
+                if (scheduleDates != null) {
+                    nextDate = scheduleDates.getDate();
+                } else {
+                    return null;
+                }
+            }
         }
 
```

Comment:
REmove blank line so we clearly see that this is connected to the previous if branch